File: scala/p0001.scala
import scala.collection.mutable.HashMap

object Solution {
    def twoSum(nums: Array[Int], target: Int): Array[Int] = {
        var hashMap = new HashMap[Int, Int]
        for (i <- 0 until nums.size) {
            if (hashMap contains (target - nums(i))) {
                return Array(i, hashMap(target - nums(i)))
            }
            hashMap(nums(i)) = i
        }
        return Array(-1, -1)
    }
}

File: p0001.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        unordered_map<int, int> prevNums;
        for (int i = 0; i < (int)nums.size(); i++)
        {
            if (prevNums.find(target - nums[i]) != prevNums.end())
            {
                return {prevNums[target - nums[i]], i};
            }
            prevNums[nums[i]] = i;
        }
        return {-1, -1};
    }
};